src/components/ButtonWithConfirm/ButtonWithConfirm.js

```diff
@@ -99,90 +73,50 @@ class ButtonWithConfirm extends Component<Props, ComponentState> {
   }
 
   resetCountdown = () => {
-    this.finishCountdown()
-    this.setCountdown(this.props.confirmTimeout)
-    this.props.onCancelClick()
-  }
+    const {
+      onCancel,
+      confirmTimeout,
+    }: Props = this.props
 
-  handleClickInit = () => {
-    this.setState({ isActive: true })
-    this.startCountdown()
+    this.finishCountdown()
+    this.setCountdown(confirmTimeout)
+    onCancel()
   }
 
-  handleClickCancel = () => {
-    this.setState({ isActive: false })
+  handleCancel = () => {
     this.resetCountdown()
   }
 
   render() {
     const {
-      onClick,
-      color,
-      label,
-      bgColor,
-      iconTooltipName,
-      iconTooltipColor,
       labelCancel,
       labelConfirm,
-      isReverse,
+      onConfirm: handleConfirm,
     }: Props = this.props
 
-    const {
-      countdown,
-      isActive,
-    }: ComponentState = this.state
+    const { countdown }: StateProps = this.state
+    const isConfirmDisabled: boolean = (countdown > 0)
```

Comment:
May be >=0 ?

Reply:
your suggestion means that button should be disabled when we already counted to 0